### PR TITLE
Connector configuration improvements

### PR DESCRIPTION
**Pull request by spericas**
### Description

Connector configuration improvements. Adds support to configure connector from global config (registry). Fixes a few edge cases in property mapping. Some new tests. Issue #9859.

### Documentation

Included in this PR.

### CODE REVIEW

File: jersey/connector/src/main/java/io/helidon/jersey/connector/HelidonConnector.java [hunk 1]
```diff
@@ -78,42 +77,60 @@ class HelidonConnector implements Connector {
...
-            builder.connectTimeout(Duration.ofMillis(getValue(properties, CONNECT_TIMEOUT, DEFAULT_TIMEOUT)));
+            Integer connectTimeout = getValue(properties, CONNECT_TIMEOUT, Integer.class);
+            if (connectTimeout != null) {
+                builder.connectTimeout(Duration.ofMillis(connectTimeout));
+            }
         }
         if (properties.containsKey(READ_TIMEOUT)) {
-            builder.readTimeout(Duration.ofMillis(getValue(properties, READ_TIMEOUT, DEFAULT_TIMEOUT)));
+            Integer readTimeout = getValue(properties, READ_TIMEOUT, Integer.class);
+            if (readTimeout != null) {
```
Comment (ljnelson) [inline jersey/connector/src/main/java/io/helidon/jersey/connector/HelidonConnector.java:111]: (If you're going to always call `getValue` and check to see whether it worked by checking for `null`, then you can eliminate the `containsKey` call.)

Comment (spericas) [reply jersey/connector/src/main/java/io/helidon/jersey/connector/HelidonConnector.java:111]: If it's there it can still return `null` if `getValue` is unable to convert the value to the type requested. In fact, it should at least log that condition and it isn't yet.

Comment (ljnelson) [reply jersey/connector/src/main/java/io/helidon/jersey/connector/HelidonConnector.java:111]: OK!

**Review by Verdent — APPROVED** [review]
